src/arch.h

```diff
@@ -23,19 +23,18 @@ typedef unsigned short u16;
 typedef unsigned int u32;
 typedef unsigned long long u64;
 
-static inline u64 atomicInc(volatile u64& var, u64 increment = 1) {
+template <typename T>
+static inline T atomicInc(volatile T& var, T increment = 1) {
     return __sync_fetch_and_add(&var, increment);
 }
 
-static inline int atomicInc(volatile int& var, int increment = 1) {
-    return __sync_fetch_and_add(&var, increment);
-}
-
-static inline u64 loadAcquire(u64& var) {
+template <typename T>
+static inline T loadAcquire(T& var) {
     return __atomic_load_n(&var, __ATOMIC_ACQUIRE);
 }
 
-static inline void storeRelease(u64& var, u64 value) {
+template <typename T>
+static inline void storeRelease(T& var, T value) {
```

Comment:
These changes don't seem to be related to the log level handling.
Can they be left out?

Reply:
I agree that this change looks unrelated to the logging stuff.

I don't mind having `storeRelease` template for arbitrary types, but adding this would make sense with replacing all other `__atomic_store_n` usages with this template (which, I think, is better to do in a separate PR).

Reply:
I've reverted the template on `storeRelease` and replaced the use with `__atomic_store_n`.